app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/SelectionInteractionContentViewModel.kt

```diff
@@ -20,5 +21,13 @@ class SelectionInteractionContentViewModel(
     if (isCurrentlySelected != shouldNowBeSelected) {
       isAnswerSelected.set(shouldNowBeSelected)
     }
+    if (
+      selectionInteractionViewModel.selectedItems.size ==
+      selectionInteractionViewModel.maxAllowableSelectionCount
+    ) {
+      enabledItem.set(isAnswerSelected.get())
+    } else {
+      enabledItem.set(true)
+    }
   }
```

Comment:
This part should be happening in SelectionInteractionViewModel, not here.

Reply:
I think Ben, that it should be here because `SelectionInteractionViewModel` has a different XML file that implements the textView that should be changed according to the selection that the user made i.e `selection_iteraction_item.xml` and `SelectionInteractionContentViewModel` has XML `multiple_choice_interaction_items.xml` which implements Radio Buttons.

Reply:
To clarify: SelectionInteractionViewModel is responsible for managing the states of all checkboxes/radio buttons being shown. It's also a dependency inversion to have this model depend on SelectionInteractionViewModel as SelectionInteractionViewModel creates instances of this model (I'm aware that it's an existing dependency, but it ought not to be so we shouldn't continue to propagate it). Since SelectionInteractionViewModel already manages checkbox checked states, it makes sense for it to also manage enabled states.

I agree that this view model should have the ObservableBoolean for enabled that's bound via XML, but doing it in this way (calling into SelectionInteractionViewModel) is where the issue is.